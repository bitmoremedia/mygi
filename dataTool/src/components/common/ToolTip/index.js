import React, { PropTypes } from 'react'

import { Container } from './styled-components'

const ToolTip = ({ children, text, position, length }) =>
  <Container>
    <span data-balloon={text} data-balloon-pos={position} data-balloon-length={length}>
      {children}
    </span>
  </Container>

ToolTip.propTypes = {
  text: PropTypes.string.isRequired,
  position: PropTypes.oneOf(['up', 'down', 'left', 'right']),
  length: PropTypes.oneOf(['small', 'medium', 'large', 'fit']),
  children: PropTypes.any.isRequired,
}

ToolTip.defaultProps = {
  position: 'up',
  length: 'large',
}

export default ToolTip
