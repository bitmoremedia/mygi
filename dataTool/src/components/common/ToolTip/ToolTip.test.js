import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import ToolTip from './index'

test('component renders without error', () => {
  const wrapper = mount(
    <ToolTip text="tool-tip-text">
      Content
    </ToolTip>,
  )
  expect(wrapper.exists()).toBe(true)
})

// INTERACTIVITY TESTS

// SNAPSHOT TESTS

test('component renders as expected', () => {
  const component = renderer.create(
    <ToolTip text="tool-tip-text">
      Content
    </ToolTip>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
