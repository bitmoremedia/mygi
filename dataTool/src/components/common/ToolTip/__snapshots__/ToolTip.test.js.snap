exports[`test component renders as expected 1`] = `
<span
  className="kpQwSz">
  <span
    data-balloon="tool-tip-text"
    data-balloon-length="large"
    data-balloon-pos="up">
    Content
  </span>
</span>
`;
